pgwidget/pgwidget_core.py

```diff
@@ -152,11 +181,12 @@ def cursor_row_index(self):
     def cursor_row_index(self,cursor_row_index):
         if cursor_row_index is not None:
             self._cursor_row_index = max(cursor_row_index, 0)
-            self.shown_cursor_row_index=self._cursor_row_index
+            self.shown_cursor_row_index = self._cursor_row_index - self.shown_text_first_line_index
         else:
             self._cursor_row_index = None
         
-        self._recalculate_cursor_position() #TODO: add on_drag 
+        if self.shown_cursor_row_index is not None and 0 <= self.shown_cursor_row_index < len(self.shown_text_rows):
```

Comment:
0 <= self.shown_cursor_row_index < len(self.shown_text_rows)
????? what's wrong here? :)

Reply:
please also add a comment explaining the condition here

Reply:
do the same with other occurence of the same "if" above